crates/red_knot_python_semantic/resources/mdtest/type_of/any.md

```diff
@@ -0,0 +1,46 @@
+# type[Any]
+
+## Simple
+
+```py
+def f(x: type[Any]):
+    reveal_type(x)  # revealed: type[Any]
+    # TODO: could be `<object.__repr__ type> & Any`
+    reveal_type(x.__repr__)  # revealed: Any
+
+class A: ...
+
+x: type[Any] = object
+x: type[Any] = type
+x: type[Any] = A
+```
+
+## Bare type
+
+```py
+def f(x: type):
+    reveal_type(x)  # revealed: type[Any]
+    # TODO: could be `<object.__repr__ type> & Any`
+    reveal_type(x.__repr__)  # revealed: Any
```

Comment:
Per discussion this morning in Discord:
```suggestion
## Bare type

The interpretation of bare `type` is not clear: existing wording in the spec does not match the behavior of mypy or pyright. For now we interpret it as simply "an instance of `builtins.type`", which is equivalent to `type[object]`. This is similar to the current behavior of mypy, and pyright in strict mode.

```py
def f(x: type):
    reveal_type(x)  # revealed: type
    # TODO bound method types
    reveal_type(x.__repr__)  # revealed: Literal[__repr__]
```